jdaviz/configs/default/plugins/collapse/tests/test_collapse.py

```diff
@@ -1,56 +1,29 @@
 import numpy as np
 import pytest
-
-from glue.core import Data
-from jdaviz import Application
+from astropy import units as u
+from specutils import Spectrum1D
 
 from jdaviz.configs.default.plugins.collapse.collapse import Collapse
 
 
 @pytest.mark.filterwarnings('ignore')
-def test_linking_after_collapse(spectral_cube_wcs):
+def test_linking_after_collapse(cubeviz_helper, spectral_cube_wcs):
+    cubeviz_helper.load_data(Spectrum1D(flux=np.ones((3, 4, 5)) * u.nJy, wcs=spectral_cube_wcs),)
+    dc = cubeviz_helper.app.data_collection
 
-    app = Application()
-    dc = app.data_collection
-    dc.append(Data(x=np.ones((3, 4, 5)), label='test', coords=spectral_cube_wcs))
+    coll = Collapse(app=cubeviz_helper.app)
 
-    coll = Collapse(app=app)
+    coll.selected_data_item = 'Unknown spectrum object[FLUX]'
 
-    coll.selected_data_item = 'test'
+    # This is automatically called in notebook but not in this test.
```

Comment:
I don't know if this is frontend weirdness in unit test or a bug. Seems to work find in the plugin but please double check!

Reply:
Is the intention here to update the populated spectral bounds?  I think that's scheduled to be removed in the future anyways, but I am seeing it not being populated at start in this PR (but populated at start in main)...

Reply:
Correct. Because it is my impression that this would automatically be picked up by `@observe`. So the way I refactored it, setting, `self.selected_data_item` actually calls `self.selected_subset` at the end, which I thought would kick off `self._on_subset_selected()` automatically to populate the bounds. Am I misunderstanding how `@observe` works?

Reply:
Yes, except the default values set when creating the traitlets don't trigger those.  Your setup should work if you have the traitlet default to an empty string instead of `"None"` and let it update to `"None"` (and the bounds populate) when the data is loaded.

Reply:
I wasn't aware of that subtlety. I pushed a commit. Please see if that's correct. Thanks!